### PR TITLE
Add Anubis metrics schema for local benchmark results uploading

### PR DESCRIPTION
**Pull request by gunandrose4u**
### Description
1. Add metrics.py for define the metrics schema used by Anubis
2. Add two examples (llama2 and whisper) of how to save local benchmark results following Anubis metrics schema

### CODE REVIEW

File: onnxruntime/python/tools/transformers/models/llama/benchmark_all.py [hunk 1]
```diff
@@ -211,14 +223,52 @@ def save_results(results, filename):
...
     df["Batch Size"] = df["Batch Size"].astype("int")
     df["Sequence Length"] = df["Sequence Length"].astype("int")
     df["Latency (s)"] = df["Latency (s)"].astype("float")
     df["Latency (ms)"] = df["Latency (ms)"].astype("float")
     df["Throughput (tps)"] = df["Throughput (tps)"].astype("float")
     df["Memory (GB)"] = df["Memory (GB)"].astype("float")
 
-    df.to_csv(filename, index=False)
+    # get pakcage name and version
+    import pkg_resources
```
Comment (kunal-vaishnavi) [inline onnxruntime/python/tools/transformers/models/llama/benchmark_all.py:236]: For any new libraries that are imported such as `pkg_resources`, can you check that they can be imported after installing the package list in the `requirements.txt` file with `pip install -r requirements.txt`? If not, we need to add any missing packages there.

File: onnxruntime/python/tools/transformers/models/llama/benchmark_all.py [hunk 2]
```diff
@@ -211,14 +223,52 @@ def save_results(results, filename):
...
+    df["Measured Runs"] = df["Measured Runs"].astype("int")
     df["Batch Size"] = df["Batch Size"].astype("int")
     df["Sequence Length"] = df["Sequence Length"].astype("int")
     df["Latency (s)"] = df["Latency (s)"].astype("float")
     df["Latency (ms)"] = df["Latency (ms)"].astype("float")
     df["Throughput (tps)"] = df["Throughput (tps)"].astype("float")
     df["Memory (GB)"] = df["Memory (GB)"].astype("float")
 
-    df.to_csv(filename, index=False)
+    # get pakcage name and version
```
Comment (kunal-vaishnavi) [inline onnxruntime/python/tools/transformers/models/llama/benchmark_all.py:235]:
```suggestion
    # get package name and version
```

File: onnxruntime/python/tools/transformers/models/llama/benchmark_all.py [hunk 3]
```diff
@@ -211,14 +223,52 @@ def save_results(results, filename):
...
+        record.metrics.customized["latency_s_mean"] = row["Latency (s)"]
+        record.metrics.latency_ms_mean = row["Latency (ms)"]
+        record.metrics.customized["throughput_tps"] = row["Throughput (tps)"]
+        record.metrics.max_memory_usage_GB = row["Memory (GB)"]
+
+        records.append(record)
+
+    BenchmarkRecord.save_as_csv(filename, records)
+    BenchmarkRecord.save_as_json(filename.replace(".csv", ".json"), records)
+    # df.to_csv(filename, index=False)
```
Comment (kunal-vaishnavi) [inline onnxruntime/python/tools/transformers/models/llama/benchmark_all.py:271]:
```suggestion

```

File: onnxruntime/python/tools/transformers/models/whisper/benchmark_all.py [hunk 4]
```diff
@@ -266,7 +281,47 @@ def save_results(results, filename):
...
+
+        record.metrics.latency_ms_mean = row["Latency (s)"] * 1000
+        record.metrics.throughput_qps = row["Throughput (qps)"]
+        record.metrics.max_memory_usage_GB = row["Memory (GB)"]
+
+        records.append(record)
+
+    BenchmarkRecord.save_as_csv(filename, records)
+    BenchmarkRecord.save_as_json(filename.replace(".csv", ".json"), records)
+    # df.to_csv(filename, index=False)
```
Comment (kunal-vaishnavi) [inline onnxruntime/python/tools/transformers/models/whisper/benchmark_all.py:324]:
```suggestion

```

File: onnxruntime/python/tools/transformers/models/whisper/benchmark_all.py [hunk 5]
```diff
@@ -266,7 +281,47 @@ def save_results(results, filename):
     df["Memory (GB)"] = df["Memory (GB)"].astype("float")
     df["Real Time Factor (RTF)"] = df["Real Time Factor (RTF)"].astype("float")
 
-    df.to_csv(filename, index=False)
+    # get pakcage name and version
```
Comment (kunal-vaishnavi) [inline onnxruntime/python/tools/transformers/models/whisper/benchmark_all.py:284]:
```suggestion
    # get package name and version
```

Comment (ironmangithub) [reply onnxruntime/python/tools/transformers/models/whisper/benchmark_all.py]: done.

File: onnxruntime/python/tools/transformers/models/llama/benchmark_all.py (hunk 2 above)
Comment (ironmangithub) [reply onnxruntime/python/tools/transformers/models/llama/benchmark_all.py]: done.

File: onnxruntime/python/tools/transformers/models/llama/benchmark_all.py (hunk 3 above)
Comment (ironmangithub) [reply onnxruntime/python/tools/transformers/models/llama/benchmark_all.py]: done.

File: onnxruntime/python/tools/transformers/models/whisper/benchmark_all.py (hunk 4 above)
Comment (ironmangithub) [reply onnxruntime/python/tools/transformers/models/whisper/benchmark_all.py]: done.

File: onnxruntime/python/tools/transformers/models/llama/benchmark_all.py [hunk 6]
```diff
@@ -121,11 +122,19 @@
...
+        help="Path to folder to save logs and results",
+    )
+
     args = parser.parse_args()
 
     setattr(args, "model_size", args.model_name.split("/")[-1].replace(".", "-"))  # noqa: B010
     log_folder_name = f"./{args.model_size}_{args.precision}"
-    setattr(args, "log_folder", log_folder_name)  # noqa: B010
+    if not args.log_folder:
+        args.log_folder=log_folder_name  # noqa: B010
```
[the remark this answers is not in the document]
Comment (ironmangithub) [reply onnxruntime/python/tools/transformers/models/llama/benchmark_all.py]: done.

File: onnxruntime/python/tools/transformers/models/llama/benchmark_all.py (hunk 1 above)
Comment (ironmangithub) [reply onnxruntime/python/tools/transformers/models/llama/benchmark_all.py:236]: the package is 'setuptools', it is usually installed by python by default.

File: onnxruntime/python/tools/transformers/models/whisper/benchmark_all.py [hunk 7]
```diff
@@ -452,4 +523,4 @@ def main():
 
 
 if __name__ == "__main__":
-    main()
+    main()
```
[the remark this answers is not in the document]
Comment (ironmangithub) [reply onnxruntime/python/tools/transformers/models/whisper/benchmark_all.py:526]: done.

File: onnxruntime/python/tools/transformers/models/whisper/benchmark_all.py [hunk 8]
```diff
@@ -266,7 +280,55 @@ def save_results(results, filename):
     df["Memory (GB)"] = df["Memory (GB)"].astype("float")
     df["Real Time Factor (RTF)"] = df["Real Time Factor (RTF)"].astype("float")
 
-    df.to_csv(filename, index=False)
+    # get package name and version
+    import pkg_resources
+
```
[the remark this answers is not in the document]
Comment (ironmangithub) [reply onnxruntime/python/tools/transformers/models/whisper/benchmark_all.py]: done.

**Review by kunal-vaishnavi — APPROVED** [review]

**Pull request merged**